### PR TITLE
Add score player avatar and flag to HUD

### PR DESCRIPTION
**Pull request by ItsShamed**
Addresses #22761

Had this branch for a while, but since it got brought up, I decided it would be the time to PR it.
This is just a basic implementation with basic corner radius control. More could be added later, but I think it's fine to make the scope of this PR as little as possible.

https://user-images.githubusercontent.com/39100084/222996214-aea513f9-5963-496f-bc31-e1e6e68f7b17.mp4

As mentioned in https://github.com/ppy/osu/discussions/22761#discussioncomment-5202412, I'm not going to implement the `PlayerAttributeText` in this PR, because I don't want to add another `AttributeText` and I definitely do not agree with how the current text components work right now. (or at least this is not how I would see things being implemented both from a player and dev standpoint)

### CODE REVIEW

File: osu.Game/Screens/Play/HUD/PlayerAvatar.cs [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+namespace osu.Game.Screens.Play.HUD
+{
+    public partial class PlayerAvatar : CompositeDrawable, ISerialisableDrawable
+    {
+        [SettingSource(typeof(SkinnableComponentStrings), nameof(SkinnableComponentStrings.CornerRadius), nameof(SkinnableComponentStrings.CornerRadiusDescription))]
+        public new BindableFloat CornerRadius { get; set; } = new BindableFloat
+        {
+            MinValue = 0,
+            MaxValue = 63,
```
Comment (Bruno5430) [inline osu.Game/Screens/Play/HUD/PlayerAvatar.cs:22]: I'm not a reviewer, but `MaxValue` shouldn't be 64?
If the size of the `PlayerAvatar` is 128 then the `MaxValue` of the corner radius should be half of it
(I tried it locally and it's not completely round when dragging the slider to the max, it's 2 pixels off)

**Comment by Bruno5430** [comment]
`CornerRadius` should not be 0 by default, it feels out of place with the rest of the components with those sharp corners.
If I had to chose for reference, I would go the profile overlay way
With a vector size of 120 (instead of 128) and a `CornerRadius` of 40

File: osu.Game.Tests/Skins/SkinDeserialisationTest.cs [hunk 2]
```diff
@@ -61,6 +64,8 @@ public void TestSkinnableComponentsCoveredByDeserialisationTests()
 
             foreach (string oskFile in available_skins)
             {
+                Logger.Log($"Testing file {oskFile}...");
+
```
Comment (bdach) [inline osu.Game.Tests/Skins/SkinDeserialisationTest.cs:68]:
```suggestion
```

File: osu.Game/Screens/Play/HUD/PlayerAvatar.cs [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+namespace osu.Game.Screens.Play.HUD
+{
+    public partial class PlayerAvatar : CompositeDrawable, ISerialisableDrawable
+    {
+        [SettingSource(typeof(SkinnableComponentStrings), nameof(SkinnableComponentStrings.CornerRadius), nameof(SkinnableComponentStrings.CornerRadiusDescription))]
+        public new BindableFloat CornerRadius { get; set; } = new BindableFloat
+        {
+            MinValue = 0,
+            MaxValue = 64,
+            Precision = 0.01f
```
Comment (bdach) [inline osu.Game/Screens/Play/HUD/PlayerAvatar.cs:23]: I'd argue that the values should be fractional/percentages here. The fact that the avatar has a size of 128 is completely arbitrary, and the avatar can be resized anyways. I'd do a range of [0. 0.5] and multiply that by 128 to get the final size.

File: osu.Game/Screens/Play/HUD/PlayerAvatar.cs [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+            };
+        }
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+
+            avatar.User = gameplayState.Score.ScoreInfo.User;
+            avatar.CornerRadius = CornerRadius.Value;
+            CornerRadius.BindValueChanged(e => avatar.CornerRadius = e.NewValue);
```
Comment (bdach) [inline osu.Game/Screens/Play/HUD/PlayerAvatar.cs:47]:
```suggestion
            CornerRadius.BindValueChanged(e => avatar.CornerRadius = e.NewValue, true);
```

**Review by bdach — CHANGES_REQUESTED** [review]

**Review by peppy — DISMISSED** [review]

File: osu.Game/Screens/Play/HUD/PlayerAvatar.cs [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+            };
+        }
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+
+            avatar.User = gameplayState.Score.ScoreInfo.User;
+            CornerRadius.BindValueChanged(e => avatar.CornerRadius = e.NewValue * default_size, true);
+        }
```
Comment (ItsShamed) [inline osu.Game/Screens/Play/HUD/PlayerAvatar.cs:50]: This should probably be also moved to BDL to keep consistency with 5e90b67be09b842a9fb4b7114f49e4463afdce52

**Review by peppy — DISMISSED** [review]

**Comment by bdach** [comment]
Applied one more change with the end goal of having the corner radius setting show as percentage (https://github.com/ppy/osu/pull/22797/commits/7ade525eef4a3c887de1d50123682140ee9c605a).

Seems alright otherwise.

**Review by bdach — APPROVED** [review]

**Pull request merged**